terraform/modules/data-share-service/rds.tf

```diff
@@ -9,12 +9,13 @@ resource "random_password" "rds_password" {
 }
 
 resource "aws_rds_cluster" "rds_postgres_cluster" {
-  cluster_identifier = "${var.environment}-rds-db"
-  engine             = "aurora-postgresql"
-  availability_zones = ["eu-west-2a", "eu-west-2b", "eu-west-2c"]
-  database_name      = "${var.environment}rdsdb"
-  master_username    = random_string.rds_username.result
-  master_password    = random_password.rds_password.result
+  cluster_identifier                  = "${var.environment}-rds-db"
+  engine                              = "aurora-postgresql"
+  availability_zones                  = ["eu-west-2a", "eu-west-2b", "eu-west-2c"]
+  database_name                       = "${var.environment}rdsdb"
+  master_username                     = random_string.rds_username.result
+  master_password                     = random_password.rds_password.result
+  iam_database_authentication_enabled = true
```

Comment:
We may also want to do [this](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.IAMDBAuth.html#:~:text=For%20PostgreSQL%2C%20if%20the%20IAM%20role%20(rds_iam)%20is%20added%20to%20a%20user%20(including%20the%20RDS%20master%20user)%2C%20IAM%20authentication%20takes%20precedence%20over%20password%20authentication%2C%20so%20the%20user%20must%20log%20in%20as%20an%20IAM%20user)

Reply:
I'm going to leave this until I've seen it deployed and working